phase1-coordinator/src/coordinator.rs

```diff
@@ -2455,26 +2453,10 @@ where
         let mut round = Self::load_round(&mut self.storage, current_round_height)?;
 
         tracing::debug!("Resetting round and applying storage changes");
-        if let Some(error) = round
-            .reset(&reset_action.remove_participants)
-            .into_iter()
-            .map(|action| match &action {
-                StorageAction::Remove(a) => match a.clone().try_into_locator(&self.storage) {
-                    Ok(locator) => {
-                        if self.storage.exists(&locator) {
-                            return self.storage.process(action);
-                        } else {
-                            Ok(())
-                        }
-                    }
-                    Err(e) => Err(e),
-                },
-                _ => self.storage.process(action),
-            })
-            .find_map(Result::err)
-        {
-            return Err(error);
-        }
+        self.storage.process(round.reset(&reset_action.remove_participants))?;
+
+        // Clear all files
+        self.storage.clear_round_files(current_round_height)?;
```

Comment:
I think it would be good to represent `clear_round_files` as a storage action, returned by `round.reset()` instead of calling it directly here. That better fits with the existing pattern and improves testability.